src/resolve/test-resolved-stream.c

```diff
@@ -371,10 +372,19 @@ static void try_isolate_network(void) {
         assert_se(ioctl(socket_fd, SIOCSIFFLAGS, &req) >= 0);
 }
 
+static in_port_t random_port(void) {
+        in_port_t port;
+
+        while ((port = random_u64()) <= 1024)
```

Comment:
And, should not be larger than UINT16_MAX...
Why not to simply use `random_u64_range(UINT16_MAX - 1024) + 1024` ?